src/lib/Util/adapters/bbc-kaldi/group-words-by-speakers.js

```diff
@@ -0,0 +1,122 @@
+/**
+edge cases
+- more segments then words - not an issue if you start by matching words with segment
+and handle edge case where it doesn't find a match
+- more words then segments - orphan words
+ */
+function groupWordsInParagraphsBySpeakers(words, segments) {
+  // add speakers to each word
+  const wordsWithSpeakers = addSpeakerToEachWord(words, segments.segments);
+  // group words by speakers sequentially
+  const result = groupWordsBySpeaker(wordsWithSpeakers);
+
+  return result;
+};
+
+/**
+* Add speakers to each words
+* if it doesn't have add uknw
+* @param {*} words
+* @param {*} semgnets
+*/
+function addSpeakerToEachWord(words, segments) {
+  let tmpWordsWithSpeakers =[];
+  words.forEach((word) => {
+    const tmpSpeakerSegment = findSegmentForWord(word, segments);
```

Comment:
Not a fan of 'tmp' shortening and some others i've seen in the codebase. Shortening one character but adding ambiguity isn't worth it imo. 

(obviously this isn't the best example of that haha. but think we do need a revamp on our naming conventions)

Reply:
Yeah, we have a section on code quality under contributing, https://github.com/bbc/react-transcript-editor/blob/master/CONTRIBUTING.md#code-quality

I also wanted to try adding prettier combined with eslint rules

Reply:
ok, fair I'll leave it for now, but keep in mind for future - this can also be refactored later